python/nammu/view/AtfAreaView.py

```diff
@@ -129,50 +129,78 @@ def toggle_split_arabic(self, split_orientation,
         self.repaint()
         self.controller.syntax_highlight()
 
+    def setup_edit_area_no_split(self):
+        self.container = JScrollPane(self.edit_area)
+        self.container.setRowHeaderView(self.line_numbers_area)
+        self.container.setVisible(True)
+        self.add(self.container, BorderLayout.CENTER)
+        self.controller.controller.arabic_edition_on = False
+
+        # Reset the scroll listener
+        self.vert_scroll = self.container.getVerticalScrollBar()
+        listener = atfAreaAdjustmentListener(self)
+        self.vert_scroll.addAdjustmentListener(listener)
+        menu = self.controller.controller.menuController.view.getMenu(3)
```

Comment:
I'm not particularly happy with hard-coding the position of the menu and the items (lines below).  However, for the time being I couldn't come up with a solution different from "iterate over all menus/items and find the one matching this hard-coded text" which has the same flaw.

Reply:
This doesn't fully solve the problem, but you could hide some of this in the `MenuController` (through methods like `disable_xxxx_split`). At least then it's closer to the source, and you could define the positions as constants in there.

Reply:
Actually, here is a better solution:

- In `MenuView`, keep track of the position of each menu as they are created. For example, have a dictionary `menu_positions` that maps menu names to indices, and populate it during the loop in the [constructor loop](https://github.com/oracc/nammu/blob/1ae222ca93e5f35ff4f159dd75aa7e04ebc31915/python/nammu/view/MenuView.py#L111-L117).
- Add a method to `MenuView` that gets a menu by its name. This could just be
```python
def getMenuByName(self, name):
    return self.menu_positions[name]
```
- Do something similar in each `Menu`. For example, in the constructor:
```python
self.menu_positions = {name: index for (index, name) in enumerate(menuItems)}
```
(there's a chance this doesn't work in Python 2.7, but a loop would)

- In the `MenuView` again, add a method to en/disable a particular option in a menu:
```python
def disable_item(self, menu_name, menu_item):
    menu_index = self.menu_positions[menu_name]
    menu_item_index = self.getMenu(menu_index).menu_positions[menu_item]
    self.getMenu(menu_index).getItem(menu_item_index).setEnabled(False)
```

- Then, in the `MenuController`, you can have a method like:
```python
def disable_split_options(self, horizontal=False, vertical=False, arabic=False):
    # choose the right item names here
    for option in options:
        self.view.disable_item("Window", option)
```